### PR TITLE
Support update operation

### PR DESCRIPTION
**Pull request by EthanLiou**
Hi Team,
I have a plan to use ES to store some user setting. I don't want to upload all settings to the backend once user changes just one setting. So I add a partial flag to control it.

### CODE REVIEW

**Comment by pitr** [comment]
Good looking PR, thanks @EthanLiou. My only concern is naming, perhaps `upsert` would be better than an ambiguous `partial`, what do you think? Also README should clarify that id field is required, since this code drops messages otherwise.

@dterror could you take a look?

**Comment by EthanLiou** [comment]
Hi @pitr, I provided a general way to set write_operation. I think it's better than the previous one.

File: lib/fluent/plugin/out_elasticsearch.rb [hunk 1]
```diff
@@ -154,17 +155,29 @@ def write(chunk)
...
       end
 
       if @parent_key && record[@parent_key]
-        meta['index']['_parent'] = record[@parent_key]
+        meta['_parent'] = record[@parent_key]
       end
 
-      bulk_message << meta
-      bulk_message << record
+      if @write_operation == "update" || @write_operation == "upsert"
```
Comment (pitr) [inline lib/fluent/plugin/out_elasticsearch.rb]: `case .. when` would work better here

File: lib/fluent/plugin/out_elasticsearch_dynamic.rb [hunk 2]
```diff
@@ -164,8 +164,20 @@ def write(chunk)
         host = "#{dynamic_conf['host']}:#{dynamic_conf['port']}"
       end
 
-      bulk_message[host] << meta
-      bulk_message[host] << record
+      if dynamic_conf["write_operation"] == "update" || dynamic_conf["write_operation"] == "upsert"
```
Comment (pitr) [inline lib/fluent/plugin/out_elasticsearch_dynamic.rb]: could you de-duplicate this code by extracting it into a function?

**Comment by pitr** [comment]
This is even better, will close #89.

Thanks for supporting the dynamic version as well, usually i have to ask :)

**Comment by EthanLiou** [comment]
Thanks for your suggestion:)

**Comment by pitr** [comment]
Looks great!

@dterror is on vacation, I hope is having fun!

**Comment by pitr** [comment]
v1.3.0 is released

**Comment by dterror** [comment]
Thanks guys, this looks great!

**Pull request merged**